scalyr_agent/configuration.py

```diff
@@ -3054,6 +3066,60 @@ def __get_config_or_environment_val(
                 % (param_type, param_name)
             )
 
+        return config_val
+
+    def __get_config_or_environment_val(
+        self,
+        config_object,
+        param_name,
+        param_type,
+        env_aware,
+        custom_env_name,
+        deprecated_names=None,
+    ):
+        """Returns a type-converted config param value or if not found, a matching environment value.
+
+        If the environment value is returned, it is also written into the config_object.
+
+        Currently only handles the following types (str, int, bool, float, JsonObject, JsonArray).
+        Also validates that environment variables can be correctly converted into the primitive type.
+
+        Both upper-case and lower-case versions of the environment variable will be checked.
+
+        @param config_object: The JsonObject config containing the field as a key
+        @param param_name: Parameter name
+        @param param_type: Parameter type
+        @param env_aware: If True and not defined in config file, look for presence of environment variable.
+        @param custom_env_name: If provided, will use this name to lookup the environment variable.  Otherwise, use
+            scalyr_<field> as the environment variable name. Both upper and lower case versions are tried.
+            Note: A non-empty value also automatically implies env_aware as True, regardless of it's value.
+        @param deprecated_names: List of synonym names for the *param_name* that were used in previous version but now
+        are deprecated. If the current *param_name* is not presented, then we also look for the deprecated names for
+        backward compatibility.
+
+        @return A python object representing the config param (or environment) value or None
+        @raises
+            JsonConversionException: if the config value or env value cannot be correctly converted.
+            TypeError: if the param_type is not supported.
+        """
+
+        config_val = self.__get_config_val(config_object, param_name, param_type)
+
+        # the config param is not presented by its current name, look for a deprecated name.
+        if config_val is None and deprecated_names is not None:
+            for name in deprecated_names:
+                config_val = self.__get_config_val(config_object, name, param_type)
+                if config_val:
+                    config_object.put(param_name, config_val)
+                    del config_object[name]
+                    if self.__logger:
+                        self.__logger.warning(
+                            "The configuration option {0} is deprecated, use {1} instead.".format(
```

Comment:
I wonder if we should add rate limit here (e.g. once per day), since I believe this code will be executed each time config is re-read and parsed.

Reply:
Done. Yea, I guess once per day should be fine for such messages.